app/src/main/java/org/example/ChessBoard.java

```diff
@@ -0,0 +1,32 @@
+package org.example;
+
+import org.example.ChessPieces.ChessPiece;
+import org.example.ChessPieces.ChessPieceGenerator;
+
+public class ChessBoard {
+    private ChessPiece mActivePiece;
+    private ChessPiece[][] board;
+
+    ChessBoard(String[] boardLayout) {
+
+        ChessPieceGenerator generator = new ChessPieceGenerator();
+
+        board = new ChessPiece[8][8];
+
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                String currPiece = boardLayout[i].substring(j, j + 1);
+
+                board[i][j] = generator.fromString(currPiece);
+            }
+        }
+    }
+
+    char getPieceCharFromCoords(int i, int j) {
+        if (board[i][j] != null) {
```

Comment:
`i` and `j` need to be verified that they're within [0, 8].

Instead of verifying them every time everywhere, it's better to create a `BoardPosition` class that can only be constructed with valid positions. Then, if something takes BoardPosition, it's guaranteed to be inside inside the board.